HOW_TO_DEBUG_WITH_GDB.md

```diff
@@ -0,0 +1,99 @@
+
+* IMMUTABILITY OF ALL CONDITIONS
+
+if you assume, that these values will never change during the development phase of memtest86plus AND memtest.efi is always loaded at this preferred address in qemu-system-x86_64 (which seems to be the case) then it is possible to hardcode the offset in the script (for the implementation see debug_memtest_simple.sh)
+
+* ADAPTABILITY TO DEVELOPMENT CHANGES
+
+if there is a chance, that these values WILL change during the development phase but memtest.efi is always loaded at this preferred address then the value can be read from header.S by the debug script just right before starting the debugging (for an example, see debug_memtest_full.sh)
+
+* EXPECTED ERRATIC BEHAVIOUR OF QEMU
+
+If it is expected that memtest.efi is NOT always loaded at the same address, it is inevitable to determine the actual loading address first. This approach comprises a DEBUG-build of OVMF.fd (which requires the cloning of the whole edk2-repository and manually build OVMF.fd). With this DEBUG-version of OVMF.fd it is possible to write the loading addresses of all modules into a debug.log.
+This proceeding has been tested successfully but is actually not implemented in one of the srcipts.
+
+###	Handle relocation of memtest
+
+memtest86plus relocates itself during the test procedures. As the script loads the symbol table with a given offset, debugging is only possible when the code is located at the original position. There are several ways to deal with relocation:
+
+* IGNORE RELOCATION
+
+Just ignore the fact that at a part of the time the symbols are not recognized by gdb as gdb has no information about the symbols, when memtest86plus has been relocated. It is still possible to debug the code since memtest86plus jumps sooner or later back to the original position and all precedures which are executed at one location are also executed at the other position.
+BUT: If a bug is position-dependent (i.e. it occurs only at the relocated position), you are not able to debug it.
+
+* DISABLE RELOCATION
+
+TODO: Is it possible to deactivate relocation? E.g. by outcommenting some code or setting a flag? Does it have benefits over the first approach?
+
+* FOLLOW RELOCATION
+
+If the position after relocation is expected to be always the same, then you can just load the symbol table twice. This is done in debug_memtest_simple.sh (the offsets are 0x201000 and 0x400000). If the locations can vary then the offsets must be determined dynamically ... todo: how?
+
+###	Build process
+
+The creation of the debug symbol table is realized by calling `make debug` with a modified Makefile. If a modification of the Makefile is not desired, it is also possible to imitate the build procedure directly in the debug-script. But this means to adjust the build procedure every time when the procedure of the original Makefile changes.
+
+###	Alternatives to gnome-terminal
+
+Todo: add alternatives to gnome-terminal, if gnome-terminal is not installed
```

Comment:
This is partially fixed now.

Reply:
I have made some little changes to this file. As for me, it is the final version now